extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -146,18 +147,49 @@ function SidebarWithRef(
   useLayoutEffect(() => {
     const sidebarElement = sidebarRef.current;
     const backdropElement = backdropRef.current;
-    if (!sidebarElement || !backdropElement) {
-      return;
-    }
-
-    if (!side) {
+    if (!side || !sidebarElement || !backdropElement) {
       return;
     }
 
     let sidebarAnimation;
     let backdropAnimation;
+    let keydownCallback;
+
+    const postInvisibleAnim = () => {
+      if (onAfterCloseRef.current) {
+        onAfterCloseRef.current();
+      }
+      sidebarAnimation = null;
+      backdropAnimation = null;
+      setMounted(false);
+    };
```

Comment:
Factored out `postInvisibleAnim` since it is used in both the closing animation and when the `esc` key callback.